plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/controls/querylog/QueryLogViewer.java

```diff
@@ -1088,10 +1086,24 @@ protected Composite createDialogArea(Composite parent) {
                 if (schemaTerm == null) {
                     schemaTerm = ModelMessages.controls_querylog_column_schema_name;
                 }
-                UIUtils.createLabelText(sqlDetailsPanel, catalogTerm,
-                    qmmStatementExecuteInfo.getCatalog(), SWT.BORDER | SWT.READ_ONLY, new GridData(GridData.FILL_HORIZONTAL));
-                UIUtils.createLabelText(sqlDetailsPanel, schemaTerm,
-                    qmmStatementExecuteInfo.getSchema(), SWT.BORDER | SWT.READ_ONLY, new GridData(GridData.FILL_HORIZONTAL));
+                String catalogName = qmmStatementExecuteInfo.getCatalog();
+                String schemaName = qmmStatementExecuteInfo.getSchema();
+                if (CommonUtils.isNotEmpty(catalogName) || CommonUtils.isNotEmpty(schemaName)) {
+                    UIUtils.createControlLabel(topFrame, catalogTerm);
+                    Composite container = CompositeFactory.newComposite(SWT.NONE)
+                        .layoutData(GridDataFactory.fillDefaults().create())
+                        .layout(GridLayoutFactory.fillDefaults().numColumns(3).create())
+                        .create(topFrame);
+
+                    Text catalogText = new Text(container, SWT.BORDER | SWT.READ_ONLY);
+                    catalogText.setText(CommonUtils.notEmpty(catalogName));
+                    catalogText.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+                    catalogText.setToolTipText(ModelMessages.controls_querylog_column_catalog_tip);
+
+                    UIUtils
+                        .createLabelText(container, schemaTerm, schemaName, SWT.BORDER | SWT.READ_ONLY, new GridData(GridData.FILL_HORIZONTAL))
+                        .setToolTipText(ModelMessages.controls_querylog_column_schema_tip);
```

Comment:
Such formatting looks weird.

I think this way is better
```suggestion
                    UIUtils .createLabelText(container, schemaTerm, schemaName, SWT.BORDER | SWT.READ_ONLY, new GridData(GridData.FILL_HORIZONTAL))
                        .setToolTipText(ModelMessages.controls_querylog_column_schema_tip);
```

Reply:
I disagree